api/api_signer.go

```diff
@@ -73,6 +73,33 @@ func NewSignHandlerFromSigner(signer signer.Signer) (h HTTPHandler, err error) {
 	}, nil
 }
 
+// a type meant to be unmarshalled from JSON so that there can be a
+// hostname field in the API
+type jsonSignRequest struct {
```

Comment:
This seems unnecessary. But I understand it is critical to change the API look and feel. Leave it for now but add a TODO comment.